src/components/Toolbelt/Toolbelt.js

```diff
@@ -1,22 +1,25 @@
 import React from 'react'
 import Button from '@material-ui/core/Button'
 import Tooltip from '@material-ui/core/Tooltip'
-import { func, string } from 'prop-types'
+import PropTypes from 'prop-types'
 import classNames from 'classnames'
 
 import FarmhandContext from '../../Farmhand.context'
 import './Toolbelt.sass'
 import { tools as toolImages, pixel } from '../../img'
 import toolsData from '../../data/tools'
 
-const tools = Object.values(toolsData).sort(t => t.order)
-
 const noop = () => {}
 
 export const Toolbelt = ({
   fieldMode: currentFieldMode,
   handleFieldModeSelect,
+  shovelUnlocked,
 }) => {
+  const tools = Object.values(toolsData)
+    .filter(t => shovelUnlocked || t.id !== 'shovel')
+    .sort(t => t.order)
```

Comment:
I'd say this is a warm code path, so I'd like to avoid repeating work where it's reasonable to do so. That said, this seems like a good candidate for memoization. Take a look at this util: https://github.com/jeremyckahn/farmhand/blob/cb4734a9727f2268b9cd3cbc9a1ac70a9d6d8510/src/utils.js#L123-L127

It should be pretty easy to move this calculation out of the render function and memoize it for faster re-execution. Let me know what you think of this idea!

Reply:
one of the issues it that the result of the filtering could change on any render. i originally had it outside of render, and it would not show the shovel after unlocking until completely refreshing the app, so i moved it into here. i'm sure we can come up with something, but wanted to say that in case it triggers any more thoughts for you. can chat in discord to sort this out

Reply:
Hm, that's odd. It looks like the only state this depends on is `shovelUnlocked` property. I would expect a function outside of this component that looks something like this might do it:

```js
const getTools = memoize((shovelUnlocked) => {
  return Object.values(toolsData)
    .filter(t => shovelUnlocked || t.id !== 'shovel')
    .sort(t => t.order)
})
```

And then it would be called here with `getTools(shovelUnlocked)`. You may want to try this without the `memoize` wrapper to see if the base logic works first. Feel free to ping me on DIscord and I'll respond as I can!

Reply:
yup this works! thanks for the help, i see what i had wrong now too